### PR TITLE
IPC-364: relayer docs and parent finality command

### LINKED ISSUE #364: Add docs for leave/claim, relayers and bottom-up checkpoint commands

### PR DESCRIPTION
**Pull request by adlrocha**
Closes #364 

- Adds relayer commands. 
- Includes a `cross-msg parent-finality` command to check the latest parent finality committed in a subnet.

### CODE REVIEW

File: docs/quickstart-calibration.md [hunk 1]
```diff
@@ -189,8 +189,8 @@ With the bootstrap node deployed and advertised to the network, we are now ready
 * First we need to export the private keys of our validators from the addresses that we created with our `ipc-cli wallet` to a known path so they can be picked by Fendermint to sign blocks. We can use the default repo of IPC for this, `~/.ipc`.
 ```bash
 ./bin/ipc-cli wallet export -w evm -a <WALLET_ADDR1> --hex -o ~/.ipc/<PRIV_KEY_VALIDATOR_1>
-./bin/ipc-cli wallet export -w evm -a <WALLET_ADDR2> --hex -o ~/.ipc/<PRIV_KEY_VALIDATOR_2>
-./bin/ipc-cli wallet export -w evm -a <WALLET_ADDR3> --hex -o ~/.ipc/<PRIV_KEY_VALIDATOR_3>
+./bin/ipc-cli wallet export -w evm -a <WALLET_ADDR1> --hex -o ~/.ipc/<PRIV_KEY_VALIDATOR_1>
+./bin/ipc-cli wallet export -w evm -a <WALLET_ADDR1> --hex -o ~/.ipc/<PRIV_KEY_VALIDATOR_1>
```
Comment (aakoshh) [inline docs/quickstart-calibration.md:193]: Did you really want all of these to be the same?

File: docs/usage.md [hunk 2]
```diff
@@ -169,8 +169,15 @@ fund performed in epoch 1030279
...
 
 The epoch were the message is performed can give you a sense of the time the message will take to be propagated. You can check the current finality in a subnet and wait for the finality height that includes your message to be committed.
+```bash
+./bin/ipc-cli cross-msg parent-finality --subnet <SUBNET_ID>
+```
 
-> TODO: Add a cli command to check the finality committed in a child.
+```console
+# Example execution
+$ ./bin/ipc-cli cross-msg parent-finality --subnet /r31415926/t4xwzbdu7z5sam6hc57xxwkctciuaz7oe5omipwbq --to=0x406a7a1d002b71ece175cc7e067620ae5b58e9ec 100
```
Comment (aakoshh) [inline docs/usage.md:178]: Is that a real address? I thought it would be t410 prefix :thinking:

File: docs/quickstart-calibration.md (hunk 1 above)
Comment (adlrocha) [reply docs/quickstart-calibration.md:193]: I don't. I don't know why my IDE started changing stuff automatically. If you find any weird typos it is some extension that decided it was a good idea to auto correct md files without asking.

File: docs/usage.md [hunk 3]
```diff
@@ -210,6 +217,19 @@ release performed in epoch 1030
...
 
+The propagation of a bottom-up checkpoint from a child subnet to its parent follows these stages:
+* Validators in the child subnet populate the checkpoint, sign it, and agree on their validity. When validators have agreed on the validity of a checkpoint for a specific epoch, a new `QuorumReached` event is emitted in the child subnet. You can check if a checkpoint for certain epoch has already been signed by a majority of child validators through the following command: `./bin/ipc-cli checkpoint quorum-reached-events --from-epoch 600 --to-epoch 680 --subnet`.
+```console
+# Sample execution
+./bin/ipc-cli checkpoint quorum-reached-events --from-epoch 600 --to-epoch 680 --subnet /r314159/t410ffumhfeppdjixhkxtgagowxkdu77j7xz5aaa52vy
+```
+
+* Once validators have agree on the checkpoint to be submitted in the parent for a specific epoch, relayers need to pick up the checkpoint and submit it in the parent. The following commands can be used to determine what is the state of this submission:
```
Comment (aakoshh) [inline docs/usage.md:227]:
```suggestion
* Once validators have agreed on the checkpoint to be submitted in the parent for a specific epoch, relayers need to pick up the checkpoint and submit it in the parent. The following commands can be used to determine what is the state of this submission:
```

File: docs/usage.md [hunk 4]
```diff
@@ -210,6 +217,19 @@ release performed in epoch 1030
...
+  * Check if the address of a relayer has already submitted a checkpoint for execution in the parent: `./bin/ipc-cli checkpoint has-submitted-bottomup-height --subnet <SUBNET_ID> --submitted <RELAYER_ADDR>`
```
Comment (aakoshh) [inline docs/usage.md:228]: `has-submitted-bottomup-height --subnet <source> --submitted <relayer>` seems odd. Should height be a parameter?

File: docs/usage.md [hunk 5]
```diff
@@ -210,6 +217,19 @@ release performed in epoch 1030
...
+  * Check the height of the latest checkpoint committed in the parent: `./bin/ipc-cli checkpoint last-bottomup-checkpoint-height --subnet <SUBNET_ID>`
+
+Finally, the bundle of checkpoints and signatures populated and already signed by a child subnet for their submission to the parent can be checked through the `./bin/ipc-cli checkpoint list-bottomup-bundle` command.
```
Comment (aakoshh) [inline docs/usage.md:231]: Is a `--subnet` command missing?

File: docs/usage.md [hunk 6]
```diff
@@ -220,6 +240,23 @@ To recover some (or all) of the funds that were sent to a subnet through `pre-fu
...
 ```
 
+## Running a relayer
+IPC relies on the role of a specific type of peer on the network called the relayers that are responsible for submitting bottom-up checkpoints that have been finalized in a child subnet to its parent. This process is key for the commitment of child subnet checkpoints in the parent, and the execution of bottom-up cross-net messages. Without relayers, cross-net messages will only flow from top levels of the hierarchy to the bottom, but not the other way around.
+
+* *session* Run the relayer process for your subnet using your default address by calling:
+```bash
+./bin/ipc-cli checkpoint relayer --subnet <SUBNET_ID>
+```
+* In order to run the relayer from a different address you can use the `--submitted` flag:
```
Comment (aakoshh) [inline docs/usage.md:250]:
```suggestion
* In order to run the relayer from a different address you can use the `--submitter` flag: 
```

File: docs/quickstart-calibration.md [hunk 7]
```diff
@@ -142,10 +142,10 @@ Before we deploy the infrastructure for the subnet, we will have to bootstrap th
 ```
 
 ## Step 6: Deploy the infrastructure
-With the collateral and number of minimum validators fulfilled, the subnet is bootstrapped in the parent, and we can deploy the infrastructure.
+With the collateral and number of minimum validators fulfilled, the subnet is bootstrapped in teh parent, and we can deploy the infrastructure.
```
Comment (aakoshh) [inline docs/quickstart-calibration.md:145]:
```suggestion
With the collateral and number of minimum validators fulfilled, the subnet is bootstrapped in the parent, and we can deploy the infrastructure.
```

File: docs/quickstart-calibration.md (hunk 1 above)
Comment (adlrocha) [reply docs/quickstart-calibration.md:193]: Fixed. Thanks!

File: docs/usage.md (hunk 2 above)
Comment (adlrocha) [reply docs/usage.md:178]: Copy-past typo. The `to` shouldn't be a flag there, already fixed. Thanks.

File: docs/usage.md [hunk 8]
```diff
@@ -210,6 +217,19 @@ release performed in epoch 1030
 ```
 As with top-down messages, you can get a sense of the time that your message will take to get to the parent by looking at the epoch in which your bottom-up message was triggered (the output of the command), and listing the latest bottom-up checkpoints to see how far it is from being propagated.
 
+The propagation of a bottom-up checkpoint from a child subnet to its parent follows these stages:
+* Validators in the child subnet populate the checkpoint, sign it, and agree on their validity. When validators have agreed on the validity of a checkpoint for a specific epoch, a new `QuorumReached` event is emitted in the child subnet. You can check if a checkpoint for certain epoch has already been signed by a majority of child validators through the following command: `./bin/ipc-cli checkpoint quorum-reached-events --from-epoch 600 --to-epoch 680 --subnet`.
+```console
```
[the remark this answers is not in the document]
Comment (adlrocha) [reply docs/usage.md:222]: fixed.

File: docs/usage.md (hunk 4 above)
Comment (adlrocha) [reply docs/usage.md:228]: @cryptoAtwill, correct me if I am wrong, but this one only checks if a relayer for a specific address has submitted the checkpoint for the latest epoch, right? It doesn't query the submission for a specific epoch (thus the lack of the `epoch` flag). I was also looking for it when I was documenting the command.

File: docs/usage.md (hunk 5 above)
Comment (adlrocha) [reply docs/usage.md:231]: Here I just wanted to reference the command, not its specific use. But I will add all of the flags, as it seems confusing.

**Pull request merged**